File: Piece of cake/973. K Closest Points to Origin/main.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
   public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int K) {
        sort(points.begin(), points.end(), [](auto&& lhs, auto&& rhs) { return lhs[0] * lhs[0] + lhs[1] * lhs[1] < rhs[0] * rhs[0] + rhs[1] * rhs[1]; });
        vector<vector<int>> res;
        for (int i = 0; i < K; ++i) {
            res.emplace_back(points[i]);
        }
        return res;
    }
};

int main() { return 0; }

File: Piece of cake/973. K Closest Points to Origin/main.rs
#![allow(dead_code)]
struct Solution;
impl Solution {
    pub fn k_closest(mut points: Vec<Vec<i32>>, k: i32) -> Vec<Vec<i32>> {
        let (a, _, _) = points.select_nth_unstable_by_key(k as usize, |x| x[0] * x[0] + x[1] * x[1]);
        a.into()
    }
}

fn main() {
    println!("{:?}", Solution::k_closest(vec![vec![1,3],vec![-2,2]], 1));
    println!("{:?}", Solution::k_closest(vec![vec![3,3],vec![5,-1],vec![-2,4]], 2));
}